keps/sig-node/277-ephemeral-containers/README.md

```diff
@@ -206,29 +213,135 @@ There are no limits on the number of Ephemeral Containers that can be created in
+of scope for this proposal.
+
+### Reattaching and Restarting Ephemeral Containers
 
 One can reattach to a Ephemeral Container using `kubectl attach`. When supported
 by a runtime, multiple clients can attach to a single debug container and share
 the terminal. This is supported by Docker.
 
-### Killing Ephemeral Containers
+Ephemeral Containers will not be restarted automatically, and there is no
+method in the API to restart an Ephemeral Container.  Creators of Ephemeral
+Containers are expected to choose a new, unused name.
+
+### Configurable Security Policy
+
+The ability to add containers to a pod implies security trade offs. We've
+received the following requirements and feedback on the alpha implementation:
+
+*   Admission controllers should be able to enforce policy based on the
+    cumulative pod specification, so operations that prune information,
+    such as removing Ephemeral Containers should not be allowed.
+*   Restarting a pod is disruptive, so for reasons of operation, security,
+    and resource hygiene it should be possible to delete Ephemeral Containers
+    via the API. ([#84764]).
+*   Ephemeral Containers could allow privilege escalation greater than that
+    of the initial pod, so setting a custom security context should not be
+    allowed.
+*   Ephemeral Containers, which are initiated by humans for debugging purposes,
+    should be allowed a more permissive security context regular containers.
+    ([#53188])
+
+Security policy is a problem better solved through the existing extension
+mechanism for applying custom policy: [Admission Controllers].
+
+Cluster administrators will be expected to choose from one of the following
+mechanisms for restricting usage of ephemeral containers:
+
+*   Use RBAC to control which users are allowed to access the
+    `/ephemeralcontainers` subresource.
+*   Write or use a third-party admission controller to allow or reject
+    Pod updates that modify ephemeral containers based on the content of
+    the update.
+*   Disable the feature using the `EphemeralContainers` feature gate.
+
+This means that all ephemeral container features will be allowed in a default
+Kubernetes install.
+
+[Admission Controllers]: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/
+
+#### Specifying Security Context
+
+The initial implementation of Ephemeral Containers prohibited setting a
+`securityContext` for an Ephemeral Container. This is now explicitly allowed.
+Cluster administrators wishing to limit this feature should prevent this using
+admission control or RBAC.
+
+#### Compatibility with existing Admission Controllers
+
+Existing Admission Controllers concerned with security will need to be updated
+to examine the `securityContext` of `ephemeralContainers`. Admission Controllers
+configured to fail open (for example, by ignoring updates using the
+`/ephemeralcontainers` subresource or not checking ephemeral containers for
```

Comment:
If an admission controller is registered for pod updates, will it be triggered on subresource updates like this? Or does the admission registration need to be updated to explicitly care about the subresource?

Reply:
As long as the configuration doesn't filter out the `ephemeralcontainers` subresource it will get an `AdmissionReview` for a `v1.EphemeralContainers`. See https://github.com/kubernetes/kubernetes/issues/92557#issuecomment-718165665 for an example.